ecommerce/extensions/payment/processors/liqpay.py

```diff
@@ -0,0 +1,221 @@
+            raise InvalidSignatureError
+
+        # Raise an exception for payments that were not accepted.
+        # Consuming code should be responsible for handling and logging the exception.
+        decode_data = json.loads(b64decode(data).decode('utf-8'))
+        transaction_state = decode_data['status'].lower()
+        if transaction_state not in ('success', 'sandbox'):
+            error_code = decode_data.get('err_code')
+            logger.error(error_code, decode_data.get('err_decription'))
+            if transaction_state in ('error', 'failure') and error_code == 'order_id_duplicate':
+                raise DuplicateReferenceNumber
+            raise GatewayError
+
+        currency = decode_data.get('currency')
+        total = Decimal(decode_data.get('amount'))
+        transaction_id = decode_data.get('payment_id')
+        card_number = decode_data.get('sender_card_mask2')
+        card_type = decode_data.get('sender_card_type')
+
+        return HandledProcessorResponse(
+            transaction_id=transaction_id,
+            total=total,
+            currency=currency,
+            card_number=card_number,
+            card_type=card_type
+        )
+
+    def issue_credit(self, order_number, basket, reference_number, amount, currency):
+        """
+        Issue a credit for the specified transaction (Refund).
+
+        Arguments:
+            order_number (str): Order number of the order being refunded.
+            basket (Basket): Basket associated with the order being refunded.
+            reference_number (str): Reference number of the transaction being refunded.
+            amount (Decimal): amount to be credited/refunded
+            currency (string): currency of the amount to be credited
+
+        Returns:
+            str: Reference number of the *refund* transaction. Unless the payment processor groups related transactions,
+             this will *NOT* be the same as the `reference_number` argument.
+        """
+
+        try:
+            params = {
+                'action': 'refund',
+                'public_key': self.public_key,
+                'version': self.version,
+                'order_id': order_number,
+                'amount': str(amount),
+                'payment_id': reference_number,
+                'currency': currency,
+            }
+            refund_data = {
+                'signature': self.make_signature(params),
+                'data': b64encode(json.dumps(params).encode("utf-8")).decode("ascii"),
+            }
+            refund_url = urljoin(self.host, "request")
+            refund_response = requests.post(refund_url, data=refund_data, verify=False)
+            response = json.loads(refund_response.content.decode("utf-8"))
+
+            transaction_state = response.get('status').lower()
+            transaction_id = response.get('payment_id')
+            ppr = self.record_processor_response(response, transaction_id=transaction_id, basket=basket)
+        except:
```

Comment:
Bad practice to catch everything. Please do it more precisely. As a last option `exept Exception` is already better than your code.

Reply:
rewrote